lib/sumodev_deploy/tasks/assets.rb

```diff
@@ -19,6 +19,11 @@
       else
         logger.important "No Gruntfile.coffee or gulpfile.js found"
       end
+      if File.exists?("package.json")
+        run_locally "rm -rf temporary_node_modules; mkdir temporary_node_modules; cp package.json temporary_node_modules; cd temporary_node_modules; npm install --production"
```

Comment:
I suggest to replace `;` with `&&` like this:
```
run_locally "rm -rf temporary_node_modules && mkdir temporary_node_modules && cp package.json temporary_node_modules && cd temporary_node_modules && npm install --production"
```
This way it ensures that each command is executed only if the previous one was successful.

As for NPM,I don't really know details on how it works, but I think the approach is valid.